ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/BlockchainQueries.java

```diff
@@ -734,6 +778,19 @@ private TransactionWithMetadata transactionByHeaderAndIndex(
             worldStateArchive.getMutable(blockHeader.getStateRoot(), blockHeader.getHash()));
   }
 
+  /**
+   * Returns the world state for the corresponding block hash
+   *
+   * @param blockHash the block hash
+   * @return the world state at the block hash
+   */
+  public Optional<WorldState> getWorldState(final Hash blockHash) {
+    final long blockNumber =
```

Comment:
Again, we cannot go hash->blockNumber and do the rest of the calls.  We lose data.